nilearn/plotting/html_stat_map.py

```diff
@@ -319,8 +322,17 @@ def _json_view_data(bg_img, stat_map_img, mask_img, bg_min, bg_max, colors,
 
     # Create a base64 sprite for the background
     bg_sprite = BytesIO()
-    bg_data = _safe_get_data(bg_img, ensure_finite=True)
-    _save_sprite(bg_data, bg_sprite, bg_max, bg_min, None, 'gray', 'png')
+    bg_data = _safe_get_data(bg_img, ensure_finite=True).astype(float)
+    bg_mask = _safe_get_data(compute_brain_mask(bg_img),
```

Comment:
looks like the block 326-335 could be turned into a helper function to increase code readability.